pkg/cloud/google/machineactuator.go

```diff
@@ -252,6 +252,18 @@ func (gce *GCEClient) Create(_ context.Context, cluster *clusterv1.Cluster, mach
 	if verr := gce.validateMachine(machine, machineConfig); verr != nil {
 		return gce.handleMachineError(machine, verr, createEventAction)
 	}
+	if !isMaster(machineConfig.Roles) {
+		isNodeSet := false
+		for _, r := range machineConfig.Roles {
+			if r == gceconfigv1.NodeRole {
+				isNodeSet = true
+				break
+			}
+		}
+		if !isNodeSet {
+			machineConfig.Roles = append(machineConfig.Roles, gceconfigv1.NodeRole)
```

Comment:
Changing this in place does mean it will be persisted back to k8s if we do an update.  I don't know if we should do that - I can see the argument for both sides.